src/thread_pool.rs

```diff
@@ -263,13 +268,26 @@ impl WorkerThread {
 
     #[cold]
     pub unsafe fn steal_until(&self, latch: &SpinLatch) {
+        // If another thread stole our job when we panic, we must halt unwinding
+        // until that thread is finished using it.
+        struct PanicGuard<'a>(&'a SpinLatch);
+        impl<'a> Drop for PanicGuard<'a> {
+            fn drop(&mut self) {
+                while !self.0.probe() {
```

Comment:
It also allows you to create a guard without the macro, which *does* allow you to `mem::forget` it later.